src/run/runner/helpers/run_command_with_log_pipe.rs

```diff
@@ -6,7 +6,10 @@ use std::process::Command;
 use std::process::ExitStatus;
 use std::thread;
 
-pub fn run_command_with_log_pipe(mut cmd: Command) -> Result<ExitStatus> {
+pub fn run_command_with_log_pipe_and_callback<F: FnOnce(u32)>(
+    mut cmd: Command,
+    cb: F,
```

Comment:
we could name this to indicate that the callback will get the process_id as argument